Plugin.php

```diff
@@ -52,43 +60,167 @@ public function boot()
 
         // Register alias
         $alias = AliasLoader::getInstance();
-        $alias->alias('Debugbar', '\Barryvdh\Debugbar\Facade');
+        $alias->alias('Debugbar', \Barryvdh\Debugbar\Facades\Debugbar::class);
 
         // Register middleware
         if (Config::get('app.debugAjax', false)) {
-            $this->app['Illuminate\Contracts\Http\Kernel']->pushMiddleware('\Winter\Debugbar\Middleware\InterpretsAjaxExceptions');
+            $this->app[HttpKernelContract::class]->pushMiddleware(\Winter\Debugbar\Middleware\InterpretsAjaxExceptions::class);
         }
 
-        // Add styling
-        $addResources = function ($controller) {
-            $debugBar = $this->app->make('Barryvdh\Debugbar\LaravelDebugbar');
-            if ($debugBar->isEnabled()) {
-                $controller->addCss(url(Config::get('cms.pluginsPath', '/plugins') . '/winter/debugbar/assets/css/debugbar.css'));
-            }
-        };
-        Event::listen('backend.page.beforeDisplay', $addResources, PHP_INT_MAX);
-        Event::listen('cms.page.beforeDisplay', $addResources, PHP_INT_MAX);
+        $this->registerResourceInjection();
+
+        if (App::runningInBackend()) {
+            $this->addBackendCollectors();
+        } else {
+            $this->registerCmsTwigExtensions();
+            $this->addFrontendCollectors();
+        }
 
-        Event::listen('cms.page.beforeDisplay', function ($controller, $url, $page) {
-            // Twig extensions
+        $this->addGlobalCollectors();
+    }
+
+    /**
+     * Registers assets bundles
+     */
+    public function register()
+    {
+        CombineAssets::registerCallback(function ($combiner) {
+            $combiner->registerBundle('$/winter/debugbar/assets/less/debugbar.less');
+        });
+    }
+
+    /**
+     * Add globally available collectors
+     */
+    public function addGlobalCollectors()
+    {
+        /** @var \Barryvdh\Debugbar\LaravelDebugbar $debugBar */
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+        $modelsCollector = $this->app->make(ModelsCollector::class);
+        $debugBar->addCollector($modelsCollector);
```

Comment:
I still need to test if this collector is actually needed, but either way it needs to respect the config setting for collectors.models